rust/syntect-plugin/Makefile

```diff
@@ -4,12 +4,27 @@ XDG_CONFIG_HOME ?= $(HOME)/.config
 XI_CONFIG_DIR ?= $(XDG_CONFIG_HOME)/xi
 XI_PLUGIN_DIR ?= $(XI_CONFIG_DIR)/plugins
 
-target/release/xi-syntect-plugin:
+lang_configs = assets/*.toml
+
+out/syntect: $(lang_configs) out/gen_manifest.toml target/release/xi-syntect-plugin
+	mkdir -p out/syntect/bin
+	cp $(lang_configs) out/syntect
+	install target/release/xi-syntect-plugin out/syntect/bin
+	cp out/gen_manifest.toml out/syntect/manifest.toml
+
+out/gen_manifest.toml: Cargo.toml Cargo.lock
+	cargo run --release --example make_manifest
+
+target/release/xi-syntect-plugin: src/*.rs Cargo.toml Cargo.lock
 	cargo build --release
 
-install: manifest.toml target/release/xi-syntect-plugin
-	install -d $(XI_PLUGIN_DIR)/syntect/bin
-	install manifest.toml $(XI_PLUGIN_DIR)/syntect
-	install target/release/xi-syntect-plugin $(XI_PLUGIN_DIR)/syntect/bin
+install: out/syntect
+	mkdir -p $(XI_PLUGIN_DIR)
+	cp -r out/syntect $(XI_PLUGIN_DIR)/syntect
```

Comment:
This line will only work the first time it is run. The second time it will copy the `syntect` folder into the already existing `syntect` folder so you end up with `$(XI_PLUGIN_DIR)/syntect/syntect/bin/xi-syntect-plugin`.